### PR TITLE
[SPARK-LLAP-198] Adds HiveServer2CredentialProvider into Spark LLAP

### LINKED ISSUE #198: Adds HiveServer2CredentialProvider into Spark LLAP

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes to add Hive Server 2 credential provider.

## How was this patch tested?

Unit tests added.

Closes #198

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/hive/llap/Utils.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.llap
+
+private[spark] object Utils {
+  def classForNAme(className: String): Class[_] =
```
Comment (dongjoon-hyun) [inline src/main/scala/org/apache/spark/sql/hive/llap/Utils.scala]: `NAme` -> `Name`?

**Comment by dongjoon-hyun** [comment]
For this one, could you review and merge this, @jerryshao ?

File: src/main/scala/org/apache/spark/sql/hive/llap/Utils.scala (hunk 1 above)
Comment (HyukjinKwon) [reply src/main/scala/org/apache/spark/sql/hive/llap/Utils.scala]: Actually, `HS2JDBCWrapper.scala` has this util already. Will remove this file ..

**Comment by jerryshao** [comment]
Have we removed this code in Spark code base @dongjoon-hyun ?

**Comment by HyukjinKwon** [comment]
Not yet. Will do this soon by myself as soon as I am able. @dongjoon-hyun guided me before.

**Comment by jerryshao** [comment]
OK, sure. Please manually verify it on a secure cluster when this code in Spark is removed.

**Comment by HyukjinKwon** [comment]
@EricWohlstadter, just up to my knowledge, https://github.com/hortonworks-spark/spark-llap/pull/199#issuecomment-380916700 is true as written. Looks it's being loaded via `YARNHadoopDelegationTokenManager` by the java service mechanism from my reading. So, I added the test case with it.

BTW, Seems `spark.security.credentials.hiveserver2.enabled` has been `false` by default specifically for this one (`spark.security.credentials.{service}.enabled`'s true by default).

**Pull request merged**